joy2twist/src/joy2twist_node.cpp

```diff
@@ -39,17 +59,41 @@ void Joy2TwistNode::load_parameters()
   this->get_parameter<float>("angular_velocity_factor.fast", angular_velocity_factors_[FAST]);
   this->get_parameter<float>("angular_velocity_factor.regular", angular_velocity_factors_[REGULAR]);
   this->get_parameter<float>("angular_velocity_factor.slow", angular_velocity_factors_[SLOW]);
+
+  this->get_parameter<bool>("e_stop.present", e_stop_present_);
+  this->get_parameter<std::string>("e_stop.topic", e_stop_topic_);
+  this->get_parameter<std::string>("e_stop.reset_srv", e_stop_reset_srv_);
+  this->get_parameter<std::string>("e_stop.trigger_srv", e_stop_trigger_srv_);
+
   this->get_parameter<int>("button_index_map.axis.angular_z", button_index_.angular_z);
   this->get_parameter<int>("button_index_map.axis.linear_x", button_index_.linear_x);
   this->get_parameter<int>("button_index_map.axis.linear_y", button_index_.linear_y);
   this->get_parameter<int>("button_index_map.dead_man_switch", button_index_.dead_man_switch);
   this->get_parameter<int>("button_index_map.fast_mode", button_index_.fast_mode);
   this->get_parameter<int>("button_index_map.slow_mode", button_index_.slow_mode);
+  this->get_parameter<int>("button_index_map.e_stop_reset", button_index_.e_stop_reset);
+  this->get_parameter<int>("button_index_map.e_stop_trigger", button_index_.e_stop_trigger);
+  this->get_parameter<int>(
+    "button_index_map.enable_e_stop_reset", button_index_.enable_e_stop_reset);
 }
 
+void Joy2TwistNode::e_stop_cb(const MsgBool::SharedPtr bool_msg) { e_stop_state_ = bool_msg->data; }
```

Comment:
This might be moved to the header file, but I do not have any strong opinion about it

Reply:
As this is subscriber callback and not simple getter/setter/etc. I'd opt to keep it inside the cpp file even if it's really simple.